### ISSUE TITLE
Make it possible to easily navigate between pages

### ISSUE THREAD

**Issue by oliviertassinari** [mui/mui-toolpad#705]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

As MUI is building more toolpad apps for its own internal use cases, we will increasingly need to be able to list all of these applications that are running in production to avoid creating duplicates, but also to be able to quickly find them. These use cases will likely span between different "apps". I wonder how we could improve discoverability.

### Examples 🌈

- Currently, we use Notion to kind of solve this problem: https://mui-org.notion.site/X-Support-87b6c33d1d86401b8af2ae2d5fc9cf5d
- Maybe it would make sense to be able to build an app like this one:
<img width="993" alt="Screenshot 2022-07-28 at 23 57 36" src="https://user-images.githubusercontent.com/3165635/181644318-c948e173-66cd-4b48-97c7-821646426231.png">

**Comment by prakhargupta1** [comment]
Wouldn't Folders and folder level view access for end users solve this ?

**Comment by prakhargupta1** [comment]
This one is irrelevant now, but it could be needed for navigating between pages of an app.

**Comment by Janpot** [comment]
> but it could be needed for navigating between pages of an app.

- Left side navigation allows you to switch between pages
- It's also possible to bind a page navigation action to a button, which in theory allows you to build your own overview page with links to other pages

Is there anything specific that you're thinking of that we can add on top of this?

**Comment by prakhargupta1** [comment]
No, I was just thinking if someone would want to group pages in the left sidebar or add a search if there are many apps. Also, all of these can be later looked into when there might come a need to make the side navigation customizable.

I think we can close it for now.
